commandapi-core/src/main/java/dev/jorel/commandapi/Tooltip.java

```diff
@@ -34,13 +42,42 @@
 public class Tooltip<S> {
 
 	private final S object;
-	private final String tooltip;
-	
-	private Tooltip(S object, String tooltip) {
+	private final Message tooltip;
+
+	 protected Tooltip(S object, Message tooltip) {
 		this.object = object;
 		this.tooltip = tooltip;
 	}
-	
+
+	/**
+	 * Gets the suggestion for this object
+	 * @return the suggestion for this object
+	 */
+	public S getSuggestion() {
+		return object;
+	}
+
+	/**
+	 * Gets the formatted tooltip for this object
+	 * @return the formatted tooltip for this object
+	 */
+	public Message getTooltip() {
+		return tooltip;
+	}
+
+	/**
+	 * Constructs a <code>Tooltip&lt;S&gt;</code> with a suggestion and a tooltip
+	 *
+	 * @param <S> the object that the argument suggestions use
+	 * @param object the suggestion to provide to the user
+	 * @param tooltip    the tooltip to show to the user when they hover over the
+	 *                   suggestion
+	 * @return a <code>Tooltip&lt;S&gt;</code> representing this suggestion and tooltip
+	 */
+	public static <S> Tooltip<S> ofString(S object, String tooltip) {
```

Comment:
We need an `of(S, String)` method for backwards compatibility

Reply:
Added in d2efc99